pkg/resources/controllermanager/deployment.go

```diff
@@ -243,7 +245,7 @@ func getFlags(data *resources.TemplateData) ([]string, error) {
 	featureGates := []string{"RotateKubeletServerCertificate=true"}
 
 	// starting with k8s 1.21, this is always true and cannot be toggled anymore
-	if cluster.Spec.Version.Semver().Minor() < 21 {
+	if version.Minor() < 21 {
```

Comment:
This is kind of out of scope, but it's gonna break if we ever get k8s 2.0. Can we make it more robust and include the major version in this check?